rego/rego.go

```diff
@@ -501,6 +501,21 @@ func Module(filename, input string) func(r *Rego) {
 	}
 }
 
+// ParsedModule returns an argument that adds a parsed Rego module. If a string
+// module with the same filename name is added, it will override the parsed
+// module.
+func ParsedModule(module *ast.Module) func(*Rego) {
+	return func(r *Rego) {
+		var filename string
+		if module.Package.Location != nil {
+			filename = module.Package.Location.File
+		} else {
+			filename = fmt.Sprintf("module_%p.rego", module)
```

Comment:
Does this happen normally? Or is that an error case? I'm wondering if we should just raise an error instead?

Reply:
If someone constructs a module programmatically the locations won't be set. Extending the rego object to keep track of errors during instantiation and then exiting when `Eval` or `PrepareForEval` or ... are called seemed like overkill.

Reply:
Makes sense to me 👍